### PR TITLE
add "into int" behavior

### LINKED ISSUE #3273: surprising behavior of into-int
**Describe the bug**
I expected `echo "123" | into-int` to produce an integer but no output appears.
`echo 123 | into-int` behaves similarly.

**To Reproduce**
1. `nu -c 'echo "123" | into-int'`

**Expected behavior**
I expect that pipeline to yield an integer I can use for variable assignment or further processing.

**Screenshots**
Here is some example output from the `into-int` commands I am trying.
Screen Shot 2021-04-06 at 3 15 51 PM

**Configuration (please complete the following information):**

```
> version | pivot
───┬────────────────────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────
 # │      Column0       │                                                 Column1                                                  
───┼────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────
 0 │ version            │ 0.29.0                                                                                                   
 1 │ build_os           │ macos-x86_64                                                                                             
 2 │ rust_version       │ rustc 1.51.0 (2fd73fabe 2021-03-23)                                                                      
 3 │ rust_channel       │ stable-x86_64-apple-darwin (default)                                                                     
 4 │ cargo_version      │ cargo 1.51.0 (43b129a20 2021-03-16)                                                                      
 5 │ pkg_version        │ 0.29.0                                                                                                   
 6 │ build_time         │ 2021-04-06 09:50:31                                                                                      
 7 │ build_rust_channel │ release                                                                                                  
 8 │ features           │ clipboard-cli, ctrlc, default, directories, dirs, ichwh, ptree, rustyline, term, trash, uuid, which, zip 
───┴────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────
```

**Add any other context about the problem here.**
I am probably doing something wrong :).
I am trying to parse directories that follow a SemVer naming structure. 
I was hoping to convert the `1.0.0` directory name to an integer so I can compare the version numerically.

### PR DESCRIPTION
**Pull request by efx**
Close #3273.

- forcibly overwrote prior implementation
- picked up prior work, polished and renamed

### CODE REVIEW

File: crates/nu-command/src/commands/into_int.rs [hunk 1]
```diff
@@ -1,76 +1,173 @@
...
 
 pub struct IntoInt;
 
 #[derive(Deserialize)]
 pub struct IntoIntArgs {
-    pub rest: Vec<Value>,
+    pub rest: Vec<ColumnPath>,
 }
 
+#[async_trait]
```
Comment (efx) [inline crates/nu-command/src/commands/into_int.rs:17]: I cannot compile locally and am unsure why:
```
error: cannot find attribute `async_trait` in this scope
  --> crates/nu-command/src/commands/into_int.rs:17:3
   |
17 | #[async_trait]
   |   ^^^^^^^^^^^

```
Does this need to be async? 
I did not see other uses of `#[async_trait]` so would prefer to remove this.

**Comment by efx** [comment]
Any advice and tips on what I'm missing are much appreciated? I overwrote the prior `into_int` implementation with the changes from #3119 and manual renames I introduced.

File: crates/nu-command/src/commands/into_int.rs (hunk 1 above)
Comment (fdncred) [reply crates/nu-command/src/commands/into_int.rs:17]: I think Jonathan changed it yesterday where we don't use async any longer.

**Comment by fdncred** [comment]
We need to keep it named `into-int` or `into int`. We'd like to have and `into` command with a variety of other subcommands like `int`.

**Pull request merged**